tests/test_username_api.py

```diff
@@ -40,405 +46,83 @@ def get_expected_response(website, user, status):
+    self.app = username_api.app.test_client()
+    self.websites = websites
 
-  @pytest.mark.parametrize('user', data['facebook']['available_usernames'])
-  def test_available_facebook_username(self, user):
-    expected = get_expected_response('facebook', user, 404)
-    actual = get_response(self.app, 'facebook', user)
+  @parameterized.expand(load_availability_test_cases('available_username'),
+      testcase_func_name=custom_name_func)
+  def test_available_username(self, website, user):
+    expected = get_expected_response(website, user, 404)
+    actual = get_response(self.app, website, user)
 
     # this is special to facebook checking,
     # and 'profile' field is only useful for
     # frontend purposes. Skipping here.
-    del actual['profile']
-
-    message = None
-    if expected != actual:
-      message = 'The provided available username ({}) returned 200'.format(user)
-
-      user = generate_random_username(25)
-      expected = get_expected_response('facebook', user, 404)
-      actual = get_response(self.app, 'facebook', user)
-
+    if website == 'facebook':
       del actual['profile']
 
-      if expected != actual:
-        message += ' and the random username ({}) returned 200'.format(user)
-      else:
-        message += ' but {} is still available'.format(user)
-
-    if message is not None:
-      logging.getLogger().warn(message)
-
-    assert expected == actual
-
-  @pytest.mark.parametrize('user', data['openhub']['available_usernames'])
-  def test_available_openhub_username(self, user):
-    expected = get_expected_response('openhub', user, 404)
-    actual = get_response(self.app, 'openhub', user)
-
     message = None
     if expected != actual:
-      message = 'The provided available username ({}) returned 200'.format(user)
-
-      user = generate_random_username(30, first_char_letter=True)
-      expected = get_expected_response('openhub', user, 404)
-      actual = get_response(self.app, 'openhub', user)
-
-      if expected != actual:
-        message += ' and the random username ({}) returned 200'.format(user)
-      else:
-        message += ' but {} is still available'.format(user)
-
-    if message is not None:
-      logging.getLogger().warn(message)
-
-    assert expected == actual
-
-  @pytest.mark.parametrize('user', data['deviantart']['available_usernames'])
-  def test_available_deviantart_username(self, user):
-    expected = get_expected_response('deviantart', user, 404)
-    actual = get_response(self.app, 'deviantart', user)
+      username_pattern = self.websites['username_patterns'][website]
+      username_length = 6
 
-    message = None
-    if expected != actual:
-      message = 'The provided available username ({}) returned 200'.format(user)
+      user = generate_random_username(website, username_length)
+      expected = get_expected_response(website, user, 404)
+      actual = get_response(self.app, website, user)
 
-      user = generate_random_username(15)
```

Comment:
why did you change this from 15 to 6?

Reply:
15 is for `test_available_deviantart_username`. 6 is just valid minimum length.

Reply:
15 was also the minimum length, previously.  Some had higher, but 15 was the lowest value, and is the length of `zNRe3jx3isA8CoM` used in the test data.

Reply:
actually, 15 is valid maximum length.

Reply:
The random generator always produced a username of `length` previously.  Across all invocations, previously, the minimum value for length was 15.